hera_cal/frf.py

```diff
@@ -97,6 +106,312 @@ def sky_frates(uvd, keys=None, frate_standoff=0.0, frate_width_multiplier=1.0, m
+        antpairpol keys to compute fringe-rate limits for.
+    percentile_high: float, optional
+        Percent of beam-squared power above upper fringe rate.
+    to_filter: list of antpairpol tuples
+        list of antpairpol tuples of baselines to calculate fringe-rate limits for.
+    dfr: float, optional.
+        spacing of fringe-rate grid used to perform binning and percentile calc
+        in units of mHz.
+        default is None -> set to 1 / (dtime * Ntime) of uvd.
+    nfr: float, optional.
+        number of points on fringe-rate grid to perform binning and percentile calc.
+        default is None -> set to uvd.Ntimes.
+    taper: str, optional
+        taper expected for power spectrum calculations. Fringe-rates from different frequencies
+    frate_standoff: float, optional
+        Additional fringe-rate standoff in mHz to add to Omega_E b_{EW} nu/c for fringe-rate inpainting.
+        default = 0.0.
+    frate_width_multiplier: float, optional
+        fraction of horizon to fringe-rate filter.
+        default is 1.0
+    min_frate_half_width: float, optional
+        minimum fringe-rate width to filter, regardless of baseline length in mHz.
+        Default is 0.025
+    fr_freq_skip: int, optional
+        bin fringe rates from every freq_skip channels.
+        default is 1 -> takes a long time. We recommend setting this to be larger.
+    verbose: bool, optional
+        lots of text output, default is False.
+
+    Returns
+    -------
+    frate_centers: dict object,
+        Dictionary with the center fringe-rate of each baseline in to_filter in units of mHz.
+    frate_half_widths: dict object
+        Dictionary with the half widths of each fringe-rate window around the frate_centers in units of mHz.
+    """
+    frate_centers = {}
+    frate_half_widths = {}
+    if keys is None:
+        keys = uvd.get_antpairpols()
+
+    if frate_profiles is None:
+        if uvb is not None:
+            fr_grid, frate_profiles = build_fringe_rate_profiles(uvd=uvd, uvb=uvb, keys=keys, normed=True, nfr=nfr, dfr=dfr,
+                                                                 taper=taper, fr_freq_skip=fr_freq_skip, verbose=verbose)
+        else:
+            raise ValueError("Must either supply uvb or fr_profiles!")
+    else:
+        if nfr is None:
+            nfr = uvd.Ntimes
+        if dfr is None:
+            dfr = 1. / (nfr * np.mean(np.diff(np.unique(uvd.time_array))) * SDAY_KSEC)
+        fr_grid = np.arange(-nfr // 2, nfr // 2) * dfr
+
+    # get redundancies (will only compute fr-profile once for each red group).
+    unique_pols = set({})
+    for bl in keys:
+        if bl[-1] not in unique_pols:
+            unique_pols.add(bl[-1])
+    antpos, antnums = uvd.get_ENU_antpos()
+    antpos = {an: ap for an, ap in zip(antnums, antpos)}
+    reds = redcal.get_reds(antpos, pols=list(unique_pols), include_autos=True)
+    reds = [[bl for bl in rg if bl in keys or utils.reverse_bl(bl) in keys] for rg in reds]
```

Comment:
Same comment as before. Also, this is duplicated code from above and should be its own function (can be private and not unittested)